enhancements/automatic-resource-pruning.md

```diff
@@ -0,0 +1,196 @@
+---
+title: Automatic Resource Pruning
+authors:
+  - "@ryantking"
+reviewers:
+  - "@jmrodri"
+  - "@gallettilance"
+approvers:
+  - "@jmrodri"
+creation-date:2022-01-28
+last-updated: 2022-01-28
+status: implementable
+---
+
+## Summary
+
+This EP aims to provide an easy way for users to operator authors to limit the number of ephemeral resources that may
+exist on the cluster at any point in time. In this context, we define an ephemeral resource as a short-lived resource
+that is continuously created as the operator runs. Short-lived does not refer to a specific amount of time, rather, the
+fact that the resource has a defined life span. For example, a web server running in a `Pod` is not ephemeral because it
+will run until an external force such as a cluster administrator or CD pipeline acts on it while a log rotating script
+running in a `Pod` is ephemeral since it will run for until it finishes its defined work. Operator authors will be able
+to employ different strategies to limit the number of ephemeral resources such as adding a numerical limit, an age
+limit, or something custom to the author's use case.
```

Comment:
I am not sure it is the best choice for example. There is in Kubernetes the duality ReplicaSet / Job where the former starts a pod/process, which under normal conditions does not terminate whereas the later does terminate under normal conditions.
Here the resources to be cleaned may also be database backups, logs of CI runs, neither of them are running processes.